### PR TITLE
Add Additional Volume Update Integration Test

### PR DESCRIPTION
**Pull request by irishgordo**
- adding additional volumes fixture to test update
- adding additional small test to audit updating volume with new label as example

Resolves: tests/integration-volumes-update

### General:

* [X] Have you removed all sensitive information, including but not limited to access keys and passwords?
* [X] Have you checked to ensure there aren't other open or closed [Pull Requests](../../pulls) for the same bug/feature/question?

### Pull Request Guidelines:

1. [X] Does your submission pass tests?
1. [X] Have you added tests? 
1. [X] Are you addressing a single feature in this PR? 
1. [X] Are your commits atomic, addressing one change per commit?
1. [X] Are you following the conventions of the language? 
1. [] Have you saved your large formatting changes for a different PR, so we can focus on your work?
1. [X] Have you explained your rationale for why this feature is needed? 
1. [] Have you linked your PR to an [open issue](https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/)

### CODE REVIEW

**Comment by irishgordo** [comment]
Hey there,
I really dig this! :)
Saw a small opportunity to add another test case and fixture for more sanity coverage surrounding volumes to exercise the volume update ability.
I believe the linting/formatting are good.
I didn't notice any side-effects running:
```
make testunit && make testint
```
Super willing to make any adjustments or changes!
Didn't run a linter against the YAML fixture file created however.
Thanks for any input!

File: test/integration/volumes_test.go [hunk 1]
```diff
@@ -130,6 +130,30 @@ func TestWaitForVolumeLinodeIDInstance(t *testing.T) {
 	}
 }
 
+func TestUpdateVolume(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestUpdateVolume")
```
Comment (LBGarber) [inline test/integration/volumes_test.go:134]: In order for fixtures to generate correctly, maybe setup a volume here:
```go
client, volume, teardown, err := setupVolume(t, "fixtures/TestUpdateVolume")
defer teardown()
```

File: test/integration/volumes_test.go [hunk 2]
```diff
@@ -130,6 +130,30 @@ func TestWaitForVolumeLinodeIDInstance(t *testing.T) {
...
+	}
+	if volume.ID == 0 {
+		t.Errorf("Expected a volumes id, but got 0")
+	}
+	if volume.Label != "my-testing-volume" {
+		t.Errorf("Expected volume label to be equal to updated label")
+	}
+	assertDateSet(t, volume.Created)
+	assertDateSet(t, volume.Updated)
+	if err := client.DeleteVolume(context.Background(), volume.ID); err != nil {
```
Comment (LBGarber) [inline test/integration/volumes_test.go:152]: If you use `setupVolume`, the teardown will handle cleaning up the volume for us so this block won't be needed

**Review by LBGarber — CHANGES_REQUESTED** [review]
Hey there! Thanks for the awesome find and PR! :smile: 

Just a few small changes and we should be good to go! After making these changes, you should be good to run `make ARGS="-run TestUpdateVolume" fixtures` to auto-generate the fixtures YAML (make sure you strip any identifying information).

File: test/integration/volumes_test.go (hunk 2 above)
Comment (irishgordo) [reply test/integration/volumes_test.go]: definitely cleans things up :) :+1:

File: test/integration/volumes_test.go (hunk 1 above)
Comment (irishgordo) [reply test/integration/volumes_test.go]: ooo yeah, that makes a lot more sense :) :+1:

**Review by LBGarber — APPROVED** [review]
Perfect! Thanks for the great contribution! :smile:

**Comment by irishgordo** [comment]
> Hey there! Thanks for the awesome find and PR! smile
> 
> Just a few small changes and we should be good to go! After making these changes, you should be good to run `make ARGS="-run TestUpdateVolume" fixtures` to auto-generate the fixtures YAML (make sure you strip any identifying information).

Thanks for the lightening fast review! :) 
I made sure to run before pushing the last commit:
```
irishgordo@lappy:~/code/linodego$ cd test/integration/
irishgordo@lappy:~/code/linodego/test/integration$ make ARGS="-run TestUpdateVolume" fixtures
make: Nothing to be done for 'fixtures'.
```
However running:
```
irishgordo@lappy:~/code/linodego$ make ARGS="-run TestUpdateVolume" fixtures
* Running fixtures
cd ./test/integration && \
LINODE_FIXTURE_MODE="record" \
LINODE_TOKEN= \
LINODE_API_VERSION="v4beta" \
GO111MODULE="on" \
go test -timeout=60m -v -run TestUpdateVolume
2021/08/18 12:35:08 Could not find LINODE_TOKEN, please verify that it is set.
exit status 1
FAIL    test/integration        0.011s
make: *** [Makefile:49: run_fixtures] Error 1
```
did have a failure I just noticed.  
Super happy I could contribute :)

**Comment by irishgordo** [comment]
@LBGarber perhaps the auto-generation of the YAML got bork'd, I can dig into this and try to create another PR to fix it?

**Comment by LBGarber** [comment]
Oh hmm, oddly everything seems to be working correctly on my end. Let me know if you figure anything out!

**Comment by irishgordo** [comment]
@LBGarber that's a great catch!
Thanks for the help!

Once I sourced my .bashrc file with the export at the tail of it, it ran with more intent.
I did see this though:
```
irishgordo@lappy:~/code/linodego$ make ARGS="-run TestUpdateVolume" fixtures
* Running fixtures
cd ./test/integration && \
LINODE_FIXTURE_MODE="record" \
LINODE_TOKEN=awesometokenawesometokenawesometoken \
LINODE_API_VERSION="v4beta" \
GO111MODULE="on" \
go test -timeout=60m -v -run TestUpdateVolume
2021/08/18 12:47:06 [INFO] LINODE_FIXTURE_MODE record will be used for tests
=== RUN   TestUpdateVolume
    volumes_test.go:134: Error listing volumes, expected struct, got error [401] Invalid Token
    volumes_test.go:136: Error setting up volume test, [401] Invalid Token
--- FAIL: TestUpdateVolume (0.23s)
panic: runtime error: invalid memory address or nil pointer dereference
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14ebd7b]

goroutine 7 [running]:
testing.tRunner.func1.2(0x1661440, 0x23b48b0)
        /usr/lib/go-1.16/src/testing/testing.go:1144 +0x332
testing.tRunner.func1(0xc00037d380)
        /usr/lib/go-1.16/src/testing/testing.go:1147 +0x4b6
panic(0x1661440, 0x23b48b0)
        /usr/lib/go-1.16/src/runtime/panic.go:965 +0x1b9
test/integration.setupVolume.func1()
        /home/irishgordo/code/linodego/test/integration/volumes_test.go:171 +0x3b
panic(0x1661440, 0x23b48b0)
        /usr/lib/go-1.16/src/runtime/panic.go:965 +0x1b9
test/integration.TestUpdateVolume(0xc00037d380)
        /home/irishgordo/code/linodego/test/integration/volumes_test.go:142 +0x10f
testing.tRunner(0xc00037d380, 0x18eb628)
        /usr/lib/go-1.16/src/testing/testing.go:1194 +0xef
created by testing.(*T).Run
        /usr/lib/go-1.16/src/testing/testing.go:1239 +0x2b3
exit status 2
FAIL    test/integration        0.248s
make: *** [Makefile:49: run_fixtures] Error 1
```
Whereas, I just set this in my bashrc:
```
export LINODE_TOKEN="awesometokenawesometokenawesometoken"
```
But I guess that makes sense because that token is invalid.  It'd be an awesome token to have :sweat_smile:  but it couldn't be real haha.
Would I need to actually set it to a valid production token in order for that fixture Makefile hook to work?

**Comment by LBGarber** [comment]
Yep, using a valid Linode API token should do the trick! Glad you figured it out!

**Pull request merged**